### DISCUSSION TITLE
Inserting in views

### DISCUSSION THREAD

**Discussion by marcpley** [pawelsalawa/sqlitestudio#5097] [Ideas]
I know that INSERT in view isn't possible in SQLite (or SQL at all ?).
In SQLite Studio, parsing the query view to build an INSERT with the good values is quite impossible with complex WHERE clauses and JOIN clauses.
But being able to perform INSERTs only in a TABLE window is a major limitation. It limits SQLS to DB maintenance 

To get round this difficulty I propose that:
The autor of the query knows what the good INSERT values would be.
An INSERT button in a VIEW window would not fire an INSERT made by SQLS but an SQL file written by the one who write the query.
There is several solution to link the view and the SQL file. For example searching a file named "<view_name>-insert.sql"

**Comment by pawelsalawa** [comment]
SQLite has dedicated solution for it: https://sqlite.org/lang_createtrigger.html#instead_of_triggers

SQLiteStudio has planned ticket to support it (#3035), but it's been waiting out there for really long time. I just pulled it to version 3.5.0.
